parsers/lib/validation.py

```diff
@@ -5,13 +5,34 @@
 from logging import getLogger
 
 
+def check_key_is_present(datapoint, key, logger):
+    if datapoint['production'].get(key, None) is None:
+        logger.warning("Required generation type {} is missing from {}".format(
+            key, datapoint['zoneKey']), extra={'key': datapoint['zoneKey']})
+        return None
+    return True
+
+
+def check_expected_range(datapoint, value, expected_range, logger, key=None):
+    low, high = min(expected_range), max(expected_range)
+    if not (low <= value <= high):
+        key_str = 'for key `{}`'.format(key) if key else ''
+        logger.warning("{} reported total of {:.2f}MW falls outside range "
+                       "of {} {}".format(datapoint['zoneKey'], value,
+                                         expected_range, key_str),
+                       extra={'key': datapoint['zoneKey']})
+        return
```

Comment:
You might just want to return None explicitly here, but that's really a style choice.